tasks/scraping.go

```diff
@@ -0,0 +1,47 @@
+package tasks
+
+import (
+	"context"
+
+	"go-google-scraper-challenge/models"
+	"go-google-scraper-challenge/services/scraper"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/beego/beego/v2/task"
+)
+
+type ScrapingTask struct {
+	Tasker *task.Task
+}
+
+func (t *ScrapingTask) Name() string {
+	return "scraping"
+}
+
+func (t *ScrapingTask) GetTasker() *task.Task {
+	if t.Tasker == nil {
+		t.Tasker = task.NewTask(t.Name(), "0 * * * * *", perform)
+	}
+
+	return t.Tasker
+}
+
+func perform(_ context.Context) error {
+	result, err := models.GetFirstPendingResult()
+	if err != nil {
+		logs.Info("No pending result", err.Error())
+
+		return nil
+	}
+
+	logs.Info("Performing scraping task with result ID:", result.Id)
+	scraperService := scraper.Scraper{
+		Result: result,
+	}
+	err = scraperService.Run()
+	if err != nil {
+		return result.Fail()
```

Comment:
Why not return the error we get? 
```suggestion
		return err
```

Reply:
fixed in [52d25bb](https://github.com/carryall/go-google-scraper-challenge/pull/71/commits/52d25bbf4b09dcbd4ceabed478e4648b2f927d54)